### PR TITLE
Feature/scene resource

### PR DESCRIPTION
**Pull request by Dartemiss**
What's new?
- Now scenes are a resource and hence are part of the resource management flow.
- BuildOptions have been implemented to map which scenes are going to be load by default and which scenes are available on the game build.
- Refactor module scene logic to open scenes
- Remove tmp_scene now is done by UUID

### CODE REVIEW

**Review by JoanStinson — APPROVED** [review]
Naisuu

File: Engine/Helper/BuildOptions.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	uint32_t GetSceneUUID(unsigned position) const;
+	std::string GetScenePath(unsigned position) const;
+	int GetPositionFromPath(const std::string& assets_path) const;
+
+public:
+	bool is_imported = false;
+
+private:
+	std::vector<Options> build_scenes;
```
Comment (JorxSS) [inline Engine/Helper/BuildOptions.h:64]: Why don't use a Options pointer vector?

File: resource.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// Valores predeterminados siguientes para nuevos objetos
+// 
+#ifdef APSTUDIO_INVOKED
+#ifndef APSTUDIO_READONLY_SYMBOLS
+#define _APS_NEXT_RESOURCE_VALUE        101
+#define _APS_NEXT_COMMAND_VALUE         40001
+#define _APS_NEXT_CONTROL_VALUE         1001
+#define _APS_NEXT_SYMED_VALUE           101
+#endif
+#endif
```
Comment (JorxSS) [inline resource.h]: what is this?

**Review by JorxSS — APPROVED** [review]

**Comment by JorxSS** [comment]
Amazing work! GJ

File: Engine/Helper/BuildOptions.h (hunk 1 above)
Comment (Dartemiss) [reply Engine/Helper/BuildOptions.h:64]: What would be the benefit of doing so? (not sure about it)

File: resource.h (hunk 2 above)
Comment (Dartemiss) [reply resource.h]: Dafuck xD first time i see this

**Pull request merged**